### PR TITLE
[android] Minor search wheel clean up

### PR DESCRIPTION
**Pull request by arnaudvergnet**
Minor refactoring following https://github.com/organicmaps/organicmaps/pull/2905#discussion_r962525996

### CODE REVIEW

File: android/src/com/mapswithme/maps/maplayer/SearchWheel.java [hunk 1]
```diff
@@ -122,40 +115,39 @@ public SearchWheel(View frame, @NonNull View.OnClickListener onSearchPressedList
...
+    return true;
   }
 
   public void show(boolean show)
   {
     UiUtils.showIf(show, mSearchButton);
-    View searchLayout = getSearchLayout();
-    if (searchLayout != null)
-      UiUtils.showIf(show && mIsExpanded, searchLayout);
+    if (initSearchLayout() && mSearchLayout != null)
```
Comment (vng) [inline android/src/com/mapswithme/maps/maplayer/SearchWheel.java:149]: mSearchLayout != null is not needed here and below. All checks should be done in initSearchLayout

Comment (arnaudvergnet) [reply android/src/com/mapswithme/maps/maplayer/SearchWheel.java:149]: A solution would be to remove the `@Nullable` annotation from the `mSearchLayout` variable declaration. The editor would not complain and using only `if(initSearchLayout())` would not raise a warning. Which solution do you prefer?

Comment (vng) [reply android/src/com/mapswithme/maps/maplayer/SearchWheel.java:149]: Well, I think that developers are smart enough to realize a logic in this simple case without IDE warnings :)
I'd prefer not to duplicate checks in code, so remove Nullable?

File: android/src/com/mapswithme/maps/maplayer/SearchWheel.java [hunk 2]
```diff
@@ -184,7 +175,7 @@ public void onResume()
     }
 
     final String query = SearchEngine.INSTANCE.getQuery();
-    if (TextUtils.isEmpty(query))
+    if (TextUtils.isEmpty(query) || query == null)
```
Comment (vng) [inline android/src/com/mapswithme/maps/maplayer/SearchWheel.java:178]: TestUtils.isEmpty should check for null

Comment (arnaudvergnet) [reply android/src/com/mapswithme/maps/maplayer/SearchWheel.java:178]: Same issue as with `mSearchLayout`, if we remove this check the IDE will show a warning.

Comment (vng) [reply android/src/com/mapswithme/maps/maplayer/SearchWheel.java:178]: Well, I do prefer clean code instead of IDE _strange_ tips ;)

**Review by vng — APPROVED** [review]

**Pull request merged**